src/python/pants/backend/jvm/rules/targets.py

```diff
@@ -0,0 +1,808 @@
+# Copyright 2020 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from typing import Any, Dict, Iterable, Optional, Tuple, Union
+
+from pants.backend.jvm.targets.jvm_binary import JarRules
+from pants.build_graph.address import Address
+from pants.engine.target import (
+    COMMON_TARGET_FIELDS,
+    BoolField,
+    Dependencies,
+    DictStringToStringField,
+    DictStringToStringSequenceField,
+    IntField,
+    InvalidFieldTypeException,
+    PrimitiveField,
+    ProvidesField,
+    Sources,
+    StringField,
+    StringSequenceField,
+    Target,
+)
+from pants.java.jar.exclude import Exclude
+from pants.java.jar.jar_dependency import JarDependency
+from pants.util.collections import ensure_list, ensure_str_list
+from pants.util.frozendict import FrozenDict
+
+# -----------------------------------------------------------------------------------------------
+# Common JVM Fields
+# -----------------------------------------------------------------------------------------------
+
+
+class JvmExcludes(PrimitiveField):
+    """`exclude` objects to filter this target's transitive dependencies against."""
+
+    alias = "excludes"
+    value: Optional[Tuple[Exclude, ...]]
```

Comment:
I think the semantics this kind of field should have are "The value is non-optional, but it will default to `()` if not specified", whereas I think this currently reads "The value is optional, and will be `None` if not specified"?

Reply:
As discussed over DM, it is possible to do exactly what you're saying, if that makes sense for the domain.

But, I recommend sticking with this so that we have a way to distinguish between `this was not specified` vs. `this was explicitly specified to be []`. These are different:

```python
python_library()  # no sources, uses default globs
python_library(sources=[])
```

If the call site doesn't care about this nuance, it can simply check `tgt[JvmExcludes].value` to cover both notions of falsiness.